Tribler/Core/Modules/restapi/search_endpoint.py

```diff
@@ -0,0 +1,83 @@
+import json
+import logging
+
+from twisted.web import http, resource
+from Tribler.Core.Utilities.search_utils import split_into_keywords
+from Tribler.Core.exceptions import OperationNotEnabledByConfigurationException
+from Tribler.Core.simpledefs import NTFY_CHANNELCAST, NTFY_TORRENTS, SIGNAL_TORRENT, SIGNAL_ON_SEARCH_RESULTS, \
+    SIGNAL_CHANNEL
+
+
+class SearchEndpoint(resource.Resource):
+    """
+    This endpoint is responsible for searching in channels and torrents present in the local Tribler database.
+
+    A GET request to this endpoint will create a search. Results are returned over the events endpoint. First, the
+    results available in the local database will be pushed. After that, incoming Dispersy results are pushed.
+    The query to this endpoint is passed using the url, i.e. /search?q=pioneer
+
+    Example response over the events endpoint:
+    {
```

Comment:
I'm thinking maybe it would be better to have each result come individually? This way you can stream results instead of piling all of them up  and then generating a big ass event with them. It will also make it more natural as you will be receiving plenty of events containing a single result once you start getting search results from the swarm.